modules/jupyter-widget/src/create-deck.js

```diff
@@ -45,32 +45,57 @@ const jsonConverter = new deck.JSONConverter({
   configuration: jsonConverterConfiguration
 });
 
-function loadExternalLibrary({libraryName, resourceUri, onComplete}, loadResource = loadScript) {
-  // NOTE loadResource is for testing only
-  loadResource(resourceUri).then(res => {
-    const module = window[libraryName];
-    const newConfiguration = {
-      classes: extractClasses(module)
-    };
-    jsonConverter.mergeConfiguration(newConfiguration);
-    if (onComplete) onComplete();
-  });
-}
-
-function addCustomLibraries(customLibraries) {
+function addCustomLibraries(customLibraries, onComplete) {
   if (!customLibraries) {
     return;
   }
-  for (const {libraryName, resourceUri} of customLibraries) {
-    loadExternalLibrary({libraryName, resourceUri});
+  const loaded = {};
+
+  function onEachFinish() {
+    if (Object.keys(loaded).length === customLibraries.length) {
```

Comment:
What if `addCustomLibraries` is called multiple times with different parameters? It's more robust to set `loaded[libraryName] = false` before loading and check `Object.keys(loaded).every(libraryName => loaded[libraryName])` after each load.